sklearn/cluster/_dbscan_inner.pyx

```diff
@@ -5,14 +5,12 @@
 from libcpp.vector cimport vector
 cimport numpy as cnp
 
-cnp.import_array()
```

Comment:
Since we still use `cnp` in this file, it is safer to keep `cnp.import_array()`. This is the recommendation from `Cython` as well:

https://github.com/cython/cython/blob/363780535e7bfe8adb1b858feb3fccbdf3434d97/docs/examples/tutorial/numpy/convolve2.pyx#L12-L16

Reply:
Thanks for spotting this!

As of Cython>=3.0, it looks like not calling `cnp.import_array()` won't be a problem because [this piece of C-code (which should be injected at runtime](https://github.com/cython/cython/blob/363780535e7bfe8adb1b858feb3fccbdf3434d97/Cython/Utility/NumpyImportArray.c#L6-L14) takes care of calling [`import_array` from NumPy](https://docs.scipy.org/doc/numpy-1.17.0/reference/c-api.array.html#c.import_array). Still [not calling `cnp.import_array()` might raise warnings](https://github.com/cython/cython/blob/363780535e7bfe8adb1b858feb3fccbdf3434d97/tests/run/numpy_cimport_1.pyx#L23-L25), so let's keep this instruction as you propose.

Reply:
If we bring back the cnp.import_array() should we remove sklearn.clusters._dbscan_inner from the USE_NEWEST_NUMPY_C_API?

Reply:
I think not calling `cnp.import_array` is legal and doable if we only use `cnp` to access types (e.g. `cnp.float64_t`), which we should be able to here.

Moreover, `dbscan_inner` is still using NumPy array via `cnp.ndarray`. Is it possible to use memoryview instead.